ibtk/src/lagrangian/FEProjector.cpp

```diff
@@ -317,34 +320,31 @@ FEProjector::buildLumpedL2ProjectionSolver(const std::string& system_name)
         for (MeshBase::const_element_iterator el_it = el_begin; el_it != el_end; ++el_it)
         {
             const Elem* const elem = *el_it;
+#if NDEBUG && LIBMESH_VERSION_LESS_THAN(1, 5, 0)
+            {
+                auto elem_type = elem->elem_type();
+                TBOX_ASSERT(((fe_type.order == FIRST) && ((dim == 2 && (elem_type == TRI3 || elem_type == QUAD4)) ||
+                                                          (dim == 3 && (elem_type == TET4 || elem_type == HEX8)))) ||
```

Comment:
this isn't right for surface meshes.

Reply:
Actually, maybe it is - let me check.

Reply:
No you are right --- need to allow for `dim == 1` for surface-based models.

Reply:
I think we need to explicitly write out `dim` and `spacedim` or whatever libmesh calls it.

This check also forbids us from using mixed meshes - I don't know if anyone is interested in this case but we might want to support it.

Reply:
Mixed meshes are disallowed by checks in `FEMechanicsBase`.

Reply:
@drwells I think I fixed this.